sdc/datatypes/hpat_pandas_groupby_functions.py

```diff
@@ -233,6 +235,174 @@ def _sdc_pandas_groupby_generic_func_codegen(func_name, columns, func_params, de
     return func_text, global_vars
 
 
+def _sdc_pandas_groupby_sum_codegen(func_name, by_column, subject_columns,
```

Comment:
A couple of comments describing what's done in every big logical block of code in this function (e.g. filling parts, reducing them and building final result) would be useful. Alternatively this can be splitted into several functions (the one that gets input and fills all _parts lists, the one that does reduction basing on functions provided as arguments, etc), but that's probably can be done during generalizing sum to other linear operations.